File: code/1021/1021.c
#include <math.h>
#include <stdio.h>

int getCoins(double num1, double num2) {
  int n1 = round(num1 * 1000);
  int n2 = round(num2 * 1000);

  return n1 / n2;
}

double calcBanknotesAndPrint(double value, int note) {

  int numberOfBanckNotes = value / note;

  printf("%d nota(s) de R$ %d.00\n", numberOfBanckNotes, note);

  return value - (numberOfBanckNotes * note);
}

double calcCoinAndPrint(double value, double coin) {

  int numberOfCoins = getCoins(value, coin);

  printf("%d moeda(s) de R$ %.2lf\n", numberOfCoins, coin);

  return value - (numberOfCoins * coin);
}

int main() {

  double n;

  scanf("%lf", &n);

  printf("NOTAS:\n");
  n = calcBanknotesAndPrint(n, 100);
  n = calcBanknotesAndPrint(n, 50);
  n = calcBanknotesAndPrint(n, 20);
  n = calcBanknotesAndPrint(n, 10);
  n = calcBanknotesAndPrint(n, 5);
  n = calcBanknotesAndPrint(n, 2);
  printf("MOEDAS:\n");
  n = calcCoinAndPrint(n, 1.00);
  n = calcCoinAndPrint(n, 0.50);
  n = calcCoinAndPrint(n, 0.25);
  n = calcCoinAndPrint(n, 0.10);
  n = calcCoinAndPrint(n, 0.05);
  calcCoinAndPrint(n, 0.01);

  return 0;
}

File: code/1021/1021.cs
using System;

class URI
{
    static void Main(string[] args)
    {
        double monetary_value;

        monetary_value = double.Parse(Console.ReadLine());

        Console.WriteLine("NOTAS:");
        Console.WriteLine("{0} nota(s) de R$ 100.00", (int)monetary_value / 100);
        monetary_value %= 100;
        Console.WriteLine("{0} nota(s) de R$ 50.00", (int)monetary_value / 50);
        monetary_value %= 50;
        Console.WriteLine("{0} nota(s) de R$ 20.00", (int)monetary_value / 20);
        monetary_value %= 20;
        Console.WriteLine("{0} nota(s) de R$ 10.00", (int)monetary_value / 10);
        monetary_value %= 10;
        Console.WriteLine("{0} nota(s) de R$ 5.00", (int)monetary_value / 5);
        monetary_value %= 5;
        Console.WriteLine("{0} nota(s) de R$ 2.00", (int)monetary_value / 2);
        monetary_value %= 2;
        Console.WriteLine("MOEDAS:");
        Console.WriteLine("{0} moeda(s) de R$ 1.00", (int)monetary_value / 1);
        monetary_value %= 1;
        Console.WriteLine("{0} moeda(s) de R$ 0.50", (int)(monetary_value / 0.50));
        monetary_value %= 0.50;
        Console.WriteLine("{0} moeda(s) de R$ 0.25", (int)(monetary_value / 0.25));
        monetary_value %= 0.25;
        Console.WriteLine("{0} moeda(s) de R$ 0.10", (int)(monetary_value / 0.10));
        monetary_value %= 0.10;
        Console.WriteLine("{0} moeda(s) de R$ 0.05", (int)(monetary_value / 0.05));
        monetary_value %= 0.05;
        Console.WriteLine("{0} moeda(s) de R$ 0.01", (int)(monetary_value / 0.01));
    }
}